src/ngraph/runtime/backend.hpp

```diff
@@ -90,6 +91,8 @@ class ngraph::runtime::Backend
     /// \returns compiled function or nullptr on failure
     virtual std::shared_ptr<Executable> compile(std::shared_ptr<Function> func,
                                                 ngraph::pass::PassConfig& pass_config,
+                                                AllocateFunc& framework_allocator,
```

Comment:
If these optional arguments were pointers instead of references then they could default to nullptr. With references for everything it is all or nothing. You need to supply a PassConfig in order to use the allocators.